test/common/user_dedicated_admin_permissions.go

```diff
@@ -117,71 +134,282 @@ func verifyDedicatedAdminProjectPermissions(projects []projectv1.Project) bool {
+func verifyDedicatedAdminAddressPlanPermissions(t *testing.T, openshiftClient *resources.OpenshiftClient) {
+	t.Log("Verifying Dedicated admin permissions for AddressPlan Resource")
+
+	resourceName := "test-address-plan"
+
+	expectedPermission := ExpectedPermissions{
+		ExpectedCreateStatusCode: 201,
+		ExpectedReadStatusCode:   200,
+		ExpectedUpdateStatusCode: 200,
+		ExpectedDeleteStatusCode: 200,
+		ExpectedListStatusCode:   200,
+		ListPath:                 fmt.Sprintf(resources.PathListAddressPlan, AMQOnlineOperatorNamespace),
+		GetPath:                  fmt.Sprintf(resources.PathGetAddressPlan, AMQOnlineOperatorNamespace, resourceName),
+		ObjectToCreate: enmassev1beta2.AddressPlan{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: AMQOnlineOperatorNamespace,
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "AddressPlan",
+				APIVersion: "admin.enmasse.io/v1beta2",
+			},
+			Spec: enmassev1beta2.AddressPlanSpec{
+				AddressType: "queue",
+				Resources: enmassev1beta2.AddressPlanResources{
+					Router: 0.01,
+					Broker: 0.001,
+				},
+			},
+		},
 	}
 
-	// Dedicate admin can not CREATE new RHMI config
-	rhmiConfig := &integreatlyv1alpha1.RHMIConfig{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "v1alpha1",
-			Kind:       "RHMIConfig",
+	verifyCRUDLPermissions(t, openshiftClient, expectedPermission)
+}
+
+// Verify Dedicated admin permissions for AuthenticationService Resource - CRUDL
+func verifyDedicatedAdminAuthenticationServicePermissions(t *testing.T, openshiftClient *resources.OpenshiftClient) {
+	t.Log("Verifying Dedicated admin permissions for AuthenticationService Resource")
+
+	resourceName := "test-authentication-service"
+	expectedPermission := ExpectedPermissions{
+		ExpectedCreateStatusCode: 201,
+		ExpectedReadStatusCode:   200,
+		ExpectedUpdateStatusCode: 200,
+		ExpectedDeleteStatusCode: 200,
+		ExpectedListStatusCode:   200,
+		ListPath:                 fmt.Sprintf(resources.PathListAuthenticationService, AMQOnlineOperatorNamespace),
+		GetPath:                  fmt.Sprintf(resources.PathGetAuthenticationService, AMQOnlineOperatorNamespace, resourceName),
+		ObjectToCreate: enmasseadminv1beta1.AuthenticationService{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: AMQOnlineOperatorNamespace,
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "AuthenticationService",
+				APIVersion: "admin.enmasse.io/v1beta1",
+			},
+			Spec: enmasseadminv1beta1.AuthenticationServiceSpec{
+				Type: enmasseadminv1beta1.None,
+			},
 		},
 	}
-	bodyBytes, err = json.Marshal(rhmiConfig)
 
-	resp, err = openshiftClient.DoOpenshiftPostRequest(path, bodyBytes)
+	verifyCRUDLPermissions(t, openshiftClient, expectedPermission)
+}
 
-	if err != nil {
-		t.Errorf("failed to perform CREATE request for rhmi config with error : %s", err)
+func verifyDedicatedAdminAMQOnlineRolePermissions(t *testing.T, ctx *TestingContext) {
```

Comment:
@KevFan Does this check the same permissions as "verifyDedicatedAdmin...Permissions" functions?

Reply:
Kinda, this is just a check that the role & role binding is created / as expected. The actual functional CRUDL checks are missing `patch` and `watch`. I just added this in since it's part of a step (7) in the test case - https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases/blob/master/tests/authorization/b02-verify-the-permissions-of-the-users.md#check-dedicated-admin-permissions

Not sure is it needed / covered by the the other checks but added it it anyway as an in case. Can be removed if you think it's not really necessary

Reply:
Oh, I totally forgot about `patch` 😅 
Let's keep it.
Thanks for explanation.